eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php

```diff
@@ -1004,4 +1013,66 @@ protected function getHash($text)
     {
         return md5(mb_strtolower($text, 'UTF-8'));
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function archiveUrlAliasesForDeletedTranslations($locationId, $parentLocationId, array $languageCodes)
+    {
+        $parentId = $this->getRealAliasId($parentLocationId);
+
+        // filter removed Translations
+        $urlAliases = $this->listURLAliasesForLocation($locationId);
+        $removedLanguages = [];
+        foreach ($urlAliases as $urlAlias) {
+            $removedLanguages = array_merge(
+                $removedLanguages,
+                array_filter(
+                    $urlAlias->languageCodes,
+                    function ($languageCode) use ($languageCodes) {
+                        return !in_array($languageCode, $languageCodes);
+                    }
+                )
+            );
+        }
+
+        if (empty($removedLanguages)) {
+            return;
+        }
+
+        // map languageCodes to their IDs
+        $languageIds = array_map(
+            function ($languageCode) {
+                return $this->languageHandler->loadByLanguageCode($languageCode)->id;
+            },
+            $removedLanguages
+        );
+
+        $this->gateway->archiveUrlAliasesForDeletedTranslations($locationId, $parentId, $languageIds);
+    }
+
+    /**
+     * Remove corrupted URL aliases (global, custom and system).
+     *
+     * @return int Number of removed URL aliases
+     *
+     * @throws \Exception
+     */
+    public function deleteCorruptedUrlAliases()
+    {
+        $this->transactionHandler->beginTransaction();
+        try {
+            $totalCount = 0;
```

Comment:
Why `$totalCount` is intialized with 0 instead of `$this->gateway->deleteUrlAliasesWithoutLocation()` ?

Reply:
For readability.

Reply:
Fixed anyway ;)